### PR TITLE
Make Node not require scene

### PR DESCRIPTION
**Pull request by dr-vortex**
See https://forum.babylonjs.com/t/34510

### CODE REVIEW

**Review by sebavan — CHANGES_REQUESTED** [review]
I am not sure we should do this change as it creates a lot more conditions to check for all the various part of the framework. Why not creating a dummyScene in your case ?

Every objects from the scene graph is attached to a scene by design.

@RaananW what do you think ?

@RaananW what do you think ?

**Review by RaananW — COMMENTED** [review]
As must as understand the separation between the two components, a node belongs to a scene in its default state. I believe there will be consequences to offering that as a high-level API for all nodes. For example, materials are not Nodes, and do belong to a scene. When I move a node from one scene to the other, I need to take care of its dependents as well, including materials, skeletons, and many other components.

Babylon was designed this way from the beginning.

File: packages/dev/core/src/node.ts [hunk 1]
```diff
@@ -324,30 +324,40 @@ export class Node implements IBehaviorAware<Node> {
...
-     * @param scene the scene this node will be added to
+     * @param scene the scene this node will be added to (optional)
      */
-    constructor(name: string, scene: Nullable<Scene> = null) {
+    constructor(name: string, scene: Nullable<Scene> = EngineStore.LastCreatedScene) {
         this.name = name;
         this.id = name;
-        this._scene = <Scene>(scene || EngineStore.LastCreatedScene);
-        this.uniqueId = this._scene.getUniqueId();
+
```
[the remark this answers is not in the document]
Comment (dr-vortex) [reply packages/dev/core/src/node.ts:332]: Since it would require a scene

**Comment by dr-vortex** [comment]
@RaananW Please see the form thread for the closing reason.

**Pull request closed without merging**